src/main/java/com/minecolonies/coremod/entity/ai/citizen/farmer/EntityAIWorkFarmer.java

```diff
@@ -131,6 +135,46 @@ public EntityAIWorkFarmer(@NotNull final JobFarmer job)
         return BuildingFarmer.class;
     }
 
+    @Override
+    protected int getActionsDoneUntilDumping()
+    {
+        return MAX_BLOCKS_MINED;
+    }
+
+    @Override
+    protected int getActionRewardForCraftingSuccess()
+    {
+        return MAX_BLOCKS_MINED;
+    }
+
+    @Override
+    protected IAIState decide()
+    {
+        final IAIState nextState = super.decide();
+        if (nextState != START_WORKING)
+        {
+            return nextState;
+        }
+
+        if (job.getTaskQueue().isEmpty())
+        {
+            return PREPARING;
+        }
+
+        if (job.getCurrentTask() == null)
+        {
+            return PREPARING;
+        }
+
```

Comment:
Yu might need the same check here I added to the plantation.

Reply:
I believe your change was to the AbstractEntityAICrafting, which is called at the beginning here, so should cover the farmer too.

Reply:
Verify that this is indeed the case.

Reply:
Yes, I double checked